### PR TITLE
fix: Enforce project creation to access dashboard

### LINKED ISSUE #91: Mandatory Project Creation for Dashboard Access
### Hey everyone 👋 

 I've noticed that many of you are raising issues and PR's regarding actions to be taken when no project is selected.

However, I would like to outline the approach we will be following:
- We will require users to create a project if they do not already have one. This enforcement is necessary to enable them to view their project on the dashboard.

- Additionally, I encourage you to raise pull requests aimed at resolving this enforcement functionality for the user.

- Therefore, I kindly request that you keep this in mind when creating new issues or pull requests related to this topic.

Please let me know if there's anything else I can assist you with and Keep contributing 😊💙

### PR DESCRIPTION
**Pull request by ruhani2**
## What does this PR do?

Mandates users to create project for dashboard access

Fixes #91

- **New Signup** (no projects yet)
https://watch.screencastify.com/v/FDRKMV4MqzvbKPp2ySNq

- **SignIn** (no projects yet), will take care existing users without any projects
https://watch.screencastify.com/v/KM4eugqnTFFiqVlSTuRh

- **Login after project is already created**
https://watch.screencastify.com/v/2BghLyEPDc6O1TsLa13i

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Same process shown in above videos

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by ruhani2** [comment]
@hemantwasthere  can you please also add hacberfest-accepted label if it is accepted.

**Comment by hemantwasthere** [comment]
Hey @ruhani2 I saw your changes it seems good but we want to open the create project modal the moment user gets signed up and redirected to dashboard

**Review by hemantwasthere — CHANGES_REQUESTED** [review]
LGTM but can you also protect `http://app.localhost:3000/dashboard/{domain}` route so that user can't access this route as well without creating a project

**Comment by ruhani2** [comment]
> LGTM but can you also protect `http://app.localhost:3000/dashboard/{domain}` route so that user can't access this route as well without creating a project

Thank you for pointing this out. I have updated the code so that it ensures project creation for other pages within the dashboard.

Screenshot 2023-10-29 at 8 08 53 PM
Screenshot 2023-10-29 at 8 08 45 PM

Please let me know if you have any other suggestions.

File: wrappers/DashboardAccessWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const projectModal = useProjectModal()
+
+  useEffect(() => {
+    if (projects && projects.length === 0 && pathname?.includes('/dashboard')) {
+      toast.error('Accessing Review App requires a project to be created.', {
+        id: 'validate-project',
+      })
+      projectModal.openCreateProjectModal()
+    }
+  }, [pathname])
```
Comment (hemantwasthere) [inline wrappers/DashboardAccessWrapper.tsx:25]: please add `projects` in the dep array as well

**Review by hemantwasthere — CHANGES_REQUESTED** [review]

**Review by hemantwasthere — APPROVED** [review]
Looks good!

**Comment by hemantwasthere** [comment]
Thanks for your contribution @ruhani2 🚀

**Pull request merged**